graphdatascience/session/dedicated_sessions.py

```diff
@@ -74,9 +75,24 @@ def get_or_create(
         else:
             db_runner = self._create_db_runner(db_connection, neo4j_driver_options)
 
-            dbid = AuraApi.extract_id(db_connection.uri)
-            aura_db_instance = self._aura_api.list_instance(dbid)
+            aura_instance_id = (
+                db_connection.aura_instance_id
+                if db_connection.aura_instance_id
+                else AuraApi.extract_id(db_connection.uri)
+            )
 
+            aura_db_instance = self._aura_api.list_instance(aura_instance_id)
+
+            if not aura_db_instance and DbEnvironmentResolver.hosted_in_aura(db_runner):
+                if db_connection.aura_instance_id:
+                    raise ValueError(
+                        f"Aura instance with id `{db_connection.aura_instance_id}` could not be found. Please verify that the instance id is correct and that you have access to the Aura instance."
+                    )
+                else:
+                    raise ValueError(
```

Comment:
This is a strange place for this error. I think it would be more clear if we checked if the aura_instance_id variable is None